botkit.d.ts

```diff
@@ -0,0 +1,427 @@
+            history: ApiMethod;
+            info: ApiMethod;
+            invite: ApiMethod;
+            kick: ApiMethod;
+            leave: ApiMethod;
+            list: ApiMethod;
+            mark: ApiMethod;
+            open: ApiMethod;
+            rename: ApiMethod;
+            setPurpose: ApiMethod;
+            setTopic: ApiMethod;
+            unarchive: ApiMethod;
+        };
+        im: {
+            close: ApiMethod;
+            history: ApiMethod;
+            list: ApiMethod;
+            mark: ApiMethod;
+            open: ApiMethod;
+        };
+        mpim: {
+            close: ApiMethod;
+            history: ApiMethod;
+            list: ApiMethod;
+            mark: ApiMethod;
+            open: ApiMethod;
+        };
+        pins: {
+            add: ApiMethod;
+            list: ApiMethod;
+            remove: ApiMethod;
+        };
+        reactions: {
+            add: ApiMethod;
+            get: ApiMethod;
+            list: ApiMethod;
+            remove: ApiMethod;
+        };
+        rtm: {
+            start: ApiMethod;
+        };
+        search: {
+            all: ApiMethod;
+            files: ApiMethod;
+            messages: ApiMethod;
+        };
+        stars: {
+            add: ApiMethod;
+            list: ApiMethod;
+            remove: ApiMethod;
+        };
+        team: {
+            accessLogs: ApiMethod;
+            info: ApiMethod;
+        };
+        users: {
+            getPresence: ApiMethod;
+            info: ApiMethod;
+            list: ApiMethod;
+            setActive: ApiMethod;
+        };
+        callAPI(command: string, data: any, cb: (err: string | Error, response: any) => void): never;
+        callAPIWithoutToken(command: string, data: any, cb: (err: string | Error, response: any) => void): never;
+    }
+    class Bot {
+        type: string;
+        botkit: Controller;
+        config: Configuration;
+        utterances: Utterances;
+        api: WebAPI;
+        identity: {
+            id: string;
+            name: string;
+        };
+
+        constructor(controller: Controller, config: Configuration);
+        configureIncomingWebhook(options: { url: string }): Bot;
+        sendWebhook(options, cb): any;
+        configureRTM(config): Bot;
+        closeRTM(err: Error): never;
+        reconnect(err: Error): any;
+        destroy(): never;
+        startRTM(cb: (err: string | Error, bot: Bot, res) => void): Bot;
+        identifyBot(cb: (err: string | Error, data) => void): any;
+        identifyTeam(cb: (err: string | Error, id: string) => void): string;
+        startPrivateConversation(message: Message, cb: (err: Error, convo: Conversation) => void): never;
+        startConversation(message: Message, cb: (err: Error, convo: Conversation) => void): never;
+        createConversation(message: Message, cb: (err: Error, convo: Conversation) => void): never;
+        private _startDM(task: Task, user_id: string, cb: Function): any;
```

Comment:
Any reason this was prefixed with `private` ?
Although the `_` prefix does seem to imply that, [at implementation](https://github.com/howdyai/botkit/tree/master/lib/Slackbot_worker.js#L326-L338) there's no evidence that it should be enforced or that we should break usages of the method in the wild.